src/host/construction.jl

```diff
@@ -58,3 +58,19 @@ end
 
 Base.one(x::AbstractGPUMatrix{T}) where {T} = _one(one(T), x)
 Base.oneunit(x::AbstractGPUMatrix{T}) where {T} = _one(oneunit(T), x)
+
+Base.iszero(x::AbstractGPUMatrix{T}) where {T} = all(iszero, x)
+function Base.isone(x::AbstractGPUMatrix{T}) where {T}
+    n,m = size(x)
+    m != n && return false
+
+    # lazily perform `x-I`
+    bc = Broadcast.broadcasted(x, CartesianIndices(x)) do _x, inds
+        _x - (inds[1] == inds[2] ? one(_x) : zero(_x))
+    end
+    # call `GPUArrays.mapreducedim!` directly, which supports Broadcasted inputs
+    y = similar(x, Bool, 1)
+    GPUArrays.mapreducedim!(iszero, &, y, Broadcast.instantiate(bc); init=true)
+
+    Array(y)[]
```

Comment:
This is clever. May I ask why `Array(y)[]` not something like `@allow_scalar y[]`?

Reply:
No reason, it should perform the same. BenchmarkTools reveals that it allocates less, but I'm not sure why, it's supposed to be identical (performance-wise it is, the bottleneck is the memcpy API call, not the CPU allocation).